File: leetcode/problems/cpp/jewels-and-stones.cpp
// by @codeAbinash
// Time : O(n)
// Space : O(n)

#include "string"
#include "unordered_set"
using namespace std;

class Solution {
public:
   int numJewelsInStones(string jewels, string stones) {
      int count = 0;
      unordered_set<char> s;
      for (auto ch : jewels)
         s.insert(ch);
      for (auto ch : stones)
         if (s.find(ch) != s.end())
            count++;
      return count;
   }
};

File: leetcode/problems/rs/jewels-and-stones.rs
use std::collections::HashMap;

impl Solution {
    pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
        let mut map = HashMap::new();
        let mut count = 0;
        for c in jewels.chars() {
            map.insert(c, 0);
        }
        for c in stones.chars() {
            if map.contains_key(&c) {
                count += 1;
            }
        }
        count
    }
}
